sound/hda/hdac_controller.c

```diff
@@ -421,8 +421,9 @@ int snd_hdac_bus_reset_link(struct hdac_bus *bus, bool full_reset)
 	if (!full_reset)
 		goto skip_reset;
 
-	/* clear STATESTS */
-	snd_hdac_chip_writew(bus, STATESTS, STATESTS_INT_MASK);
+	/* clear STATESTS if not in reset */
+	if (snd_hdac_chip_readb(bus, GCTL) & AZX_GCTL_RESET)
+		snd_hdac_chip_writew(bus, STATESTS, STATESTS_INT_MASK);
```

Comment:
the description of the change is a bit odd. it's not clear if 

a) the driver shall skip the write to STATETS if not in reset
b) the driver shall wait until the controller is out of reset before writing to STATETS

Reply:
apart from the above commit, Im also curious to know if the error is actually something we've seen ie we're clearing the STATETS when the the controller is already in reset? And if so, can we skip the call to reset the controller in the subsequent line?